src/operator/contrib/krprod.h

```diff
@@ -0,0 +1,138 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ *  \file krprod.h
+ *  \brief Core function for Khatri-Rao product
+ *  \author Jencir Lee
+ */
+#ifndef MXNET_OPERATOR_CONTRIB_KRPROD_H_
+#define MXNET_OPERATOR_CONTRIB_KRPROD_H_
+#include <vector>
+#include "mshadow/tensor.h"
+
+namespace mxnet {
+namespace op {
+
+using namespace mshadow;
+using namespace mshadow::expr;
+
+/*!
+ * \brief Computes Khatri-Rao product
+ *
+ * Given the transpose of input matrices, this function computes the
+ * transpose of the Khatri-Rao product of the input matrices.
+ *
+ * Effectively, this implementation computes the Khatri-Rao product
+ * row by row, rather than column by column. E.g. if the input matrices
+ * are of shape (3, 2), (3, 4), (3, 5), the result matrix will be of
+ * shape (3, 2 * 4 * 5), which is (3, 40).
+ *
+ * \param out result matrix
+ * \param ts_arr vector of input matrices
+ */
+template <typename DType>
+inline void krprod
+  (Tensor<cpu, 2, DType> out,
+  const std::vector<Tensor<cpu, 2, DType> > &ts_arr) {
+  // If no input matrix, return all-one vector
```

Comment:
I can see cases where (especially if we make a Python binding to the function) users can accidentally pass an empty list (e.g. getting weight matrices corresponding to a neural net from a dict and not checking whether it is empty). Without explicit error this could lead to sneaky errors.
Conversely, I can't think of a case where we would **want** the khatri-rao product of nothing to be a vector of ones.

A happy medium could be to simply issue a warning and return a null result? The user could then deal with the null if needed?